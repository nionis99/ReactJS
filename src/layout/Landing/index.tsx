import React, { HTMLAttributes } from 'react';
import AppLogo from 'components/Logo';

interface LandingProps extends HTMLAttributes<HTMLDivElement> {
  children: React.ReactNode;
}

const Landing = ({ children, className = '', ...restProps }: LandingProps) => (
  <div
    className={`flex bg-landing bg-cover bg-center bg-no-repeat justify-center h-screen text-white p-4 ${className}`}
    {...restProps}
  >
    {children}
    <div className="flex mt-auto">
      <AppLogo width={360} height={60} />
    </div>
  </div>
);

export default Landing;
