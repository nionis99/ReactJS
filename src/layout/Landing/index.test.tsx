import React from 'react';
import renderer from 'react-test-renderer';
import Landing from '.';

it('should render landing component', () => {
  const elem = renderer
    .create(
      <Landing>
        <div>TEST</div>
      </Landing>,
    )
    .toJSON();
  expect(elem).toMatchSnapshot();
});
