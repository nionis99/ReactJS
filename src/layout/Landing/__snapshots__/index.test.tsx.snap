// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render landing component 1`] = `
<div
  className="flex bg-landing bg-cover bg-center bg-no-repeat justify-center h-screen text-white p-4 "
>
  <div>
    TEST
  </div>
  <div
    className="flex mt-auto"
  >
    <test-file-stub
      height={60}
      width={360}
    />
  </div>
</div>
`;
